x-pack/plugins/lens/public/plugin.ts

```diff
@@ -389,12 +393,19 @@ export class LensPlugin {
     if (share) {
       this.locator = share.url.locators.create(new LensAppLocatorDefinition());
 
-      share.register(
-        downloadCsvShareProvider({
-          uiSettings: core.uiSettings,
-          formatFactoryFn: () => startServices().plugins.fieldFormats.deserialize,
-        })
-      );
+      const { getStartServices } = core;
+      const startServices$ = from(getStartServices());
+      startServices$.subscribe(([, { licensing }]) => {
+        licensing?.license$.subscribe((license) => {
+          return share.register(
+            downloadCsvShareProvider({
+              uiSettings: core.uiSettings,
+              formatFactoryFn: () => startServices().plugins.fieldFormats.deserialize,
+              license,
```

Comment:
It looks like the CSV Download option is getting removed unless the license type is at least Gold. It would be better to handle that here, and use the license type as a condition to suppress registration of the option completely.

```diff
       const startServices$ = from(getStartServices());
       startServices$.subscribe(([, { licensing }]) => {
         licensing?.license$.subscribe((license) => {
-          return share.register(
-            downloadCsvShareProvider({
-              uiSettings: core.uiSettings,
-              formatFactoryFn: () => startServices().plugins.fieldFormats.deserialize,
-              license,
-            })
-          );
+          const atLeastGold = license.hasAtLeast('gold');
+          if (atLeastGold) {
+            share.register(
+              downloadCsvShareProvider({
+                uiSettings: core.uiSettings,
+                formatFactoryFn: () => startServices().plugins.fieldFormats.deserialize,
+              })
+            );
+          }
         });
       });
```

Reply:
oh I love this thank you